cumulusci/tasks/github/publish.py

```diff
@@ -0,0 +1,165 @@
+import tempfile
+from datetime import datetime
+
+import github3.exceptions
+from cumulusci.core.exceptions import GithubException
+from cumulusci.core.utils import process_bool_arg, process_list_arg
+from cumulusci.tasks.github.base import BaseGithubTask
+from cumulusci.tasks.github.util import CommitDir
+from cumulusci.utils import download_extract_github_from_repo
+
+
+class PublishSubtree(BaseGithubTask):
+    task_options = {
+        "repo_url": {"description": "The url to the public repo", "required": True},
+        "branch": {
+            "description": "The branch to update in the target repo",
+            "required": True,
+        },
+        "version": {
+            "description": "The version number to release.  Also supports latest and latest_beta to look up the latest releases.",
+            "required": True,
+        },
+        "include": {
+            "description": "A list of paths from repo root to include. Directories must end with a trailing slash."
+        },
+        "create_release": {
+            "description": "If True, create a release in the public repo.  Defaults to True"
+        },
+        "release_body": {
+            "description": "If True, the entire release body will be published to the public repo.  Defaults to False"
+        },
+        "dry_run": {
+            "description": "If True, skip creating Github data.  Defaults to False"
+        },
+    }
+
+    def _init_options(self, kwargs):
+        super()._init_options(kwargs)
+        self.options["include"] = process_list_arg(
+            self.options.get(
+                "include", ["datasets/", "documentation/", "tasks/", "unpackaged/"]
+            )
+        )
+        if self.options["version"] == "latest":
+            self.options["version"] = str(self.project_config.get_latest_version())
+        elif self.options["version"] == "latest_beta":
+            self.options["version"] = str(
+                self.project_config.get_latest_version(beta=True)
+            )
+        self.options["create_release"] = process_bool_arg(
+            self.options.get("create_release", True)
+        )
+        self.options["release_body"] = process_bool_arg(
+            self.options.get("release_body", False)
+        )
+        self.options["dry_run"] = process_bool_arg(self.options.get("dry_run", False))
+
+    def _get_target_repo_api(self):
+        gh = self.project_config.get_github_api(
+            self._target_repo_info["owner"], self._target_repo_info["name"]
+        )
+        return gh.repository(
+            self._target_repo_info["owner"], self._target_repo_info["name"]
+        )
+
+    def _run_task(self):
+        self._target_repo_info = self.project_config._split_repo_url(
+            self.options["repo_url"]
+        )
```

Comment:
I'd suggest moving this line into _get_target_repo_api and not assigning it to an instance variable.